pkg/beholder/noop.go

```diff
@@ -0,0 +1,129 @@
+package beholder
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	otellognoop "go.opentelemetry.io/otel/log/noop"
+	otelmetricnoop "go.opentelemetry.io/otel/metric/noop"
+	sdklog "go.opentelemetry.io/otel/sdk/log"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	oteltracenoop "go.opentelemetry.io/otel/trace/noop"
+)
+
+// Default client to fallback when is is not initialized properly
+func NewNoopClient() OtelClient {
+	cfg := DefaultConfig()
+	// Logger
+	loggerProvider := otellognoop.NewLoggerProvider()
+	logger := loggerProvider.Logger(defaultPackageName)
+	// Tracer
+	tracerProvider := oteltracenoop.NewTracerProvider()
+	tracer := tracerProvider.Tracer(defaultPackageName)
+
+	// Meter
+	meterProvider := otelmetricnoop.NewMeterProvider()
+	meter := meterProvider.Meter(defaultPackageName)
+
+	// MessageEmitter
+	messageEmitter := noopMessageEmitter{}
+
+	client := OtelClient{cfg, logger, tracer, meter, messageEmitter, loggerProvider, tracerProvider, meterProvider, loggerProvider, noopOnClose}
+
+	return client
+}
+
+// NewStdoutClient creates a new Client with stdout exporters
+// Use for testing and debugging
+// Also this client is used as a noop client when otel exporter is not initialized properly
+func NewStdoutClient(opts ...StddutClientOption) OtelClient {
+	cfg := DefaultStdoutClientConfig()
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	// Logger
+	loggerExporter, _ := stdoutlog.New(
```

Comment:
Unchecked error

Reply:
Oh there are a few more below. I see the implementations do not actually return errors right now, but this seems like an unnecessary risk. Let's propagate any errors up through this func.

Reply:
https://github.com/smartcontractkit/chainlink-common/pull/694/commits/4ebb5fc4fcd38e805ac6290d3d04ff7c44e9d729